SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp

```diff
@@ -36,6 +36,23 @@ namespace
 		}
 		UE_LOG(LogSpatialGDKSettings, Log, TEXT("%s is %s."), PrettyName, bOutValue ? TEXT("enabled") : TEXT("disabled"));
 	}
+
+	void CheckCmdLineOverrideOptionalBool(const TCHAR* CommandLine, const TCHAR* Parameter, const TCHAR* PrettyName, TOptional<bool>& bOutValue)
+	{
+		if (FParse::Param(CommandLine, Parameter))
+		{
+			bOutValue = true;
+		}
+		else
+		{
+			TCHAR TempStr[16];
+			if (FParse::Value(CommandLine, Parameter, TempStr, 16) && TempStr[0] == '=')
+			{
+				bOutValue = FCString::ToBool(TempStr + 1); // + 1 to skip =
+			}
+		}
+		UE_LOG(LogSpatialGDKSettings, Log, TEXT("%s is %s."), PrettyName, bOutValue ? TEXT("enabled") : TEXT("disabled"));
```

Comment:
Third state here for undefined?

Reply:
If the optional is not set the it will be in the third state no? That is what my testing suggests. The world setting is respected if the command line arg is not present.